app/products/calls/alerts.test.ts

```diff
@@ -0,0 +1,486 @@
+    hasMicrophonePermission: jest.fn().mockResolvedValue(true),
+    unmuteMyself: jest.fn(),
+    dismissIncomingCall: jest.fn(),
+    removeIncomingCall: jest.fn(),
+}));
+jest.mock('@calls/actions/permissions', () => ({
+    hasBluetoothPermission: jest.fn().mockResolvedValue(true),
+}));
+jest.mock('@calls/state', () => ({
+    getCallsState: jest.fn(),
+    getChannelsWithCalls: jest.fn(),
+    getCurrentCall: jest.fn(),
+    getCallsConfig: jest.fn(),
+    setMicPermissionsGranted: jest.fn(),
+}));
+jest.mock('@calls/actions/calls');
+jest.mock('@screens/navigation', () => ({
+    dismissBottomSheet: jest.fn(),
+}));
+jest.mock('@queries/servers/channel');
+jest.mock('@queries/servers/user', () => ({
+    getUserById: jest.fn(() => Promise.resolve({
+        username: 'user-1',
+    })),
+    getCurrentUser: jest.fn().mockResolvedValue({
+        username: 'user-1',
+        roles: 'system_user',
+    }),
+}));
+
+describe('alerts', () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    describe('showLimitRestrictedAlert', () => {
+        const intl = {
+            formatMessage: ({defaultMessage}: {defaultMessage: string}, values?: any) => {
+                if (values) {
+                    return defaultMessage.replace('{maxParticipants}', values.maxParticipants);
+                }
+                return defaultMessage;
+            },
+        };
+
+        it('shows alert for cloud starter', () => {
+            const mockAlert = jest.spyOn(Alert, 'alert');
+            showLimitRestrictedAlert({
+                limitRestricted: true,
+                maxParticipants: 8,
+                isCloudStarter: true,
+            }, intl as any);
+
+            expect(mockAlert).toHaveBeenCalledWith(
+                'This call is at capacity',
+                'Upgrade to Cloud Professional or Cloud Enterprise to enable group calls with more than 8 participants.',
+                [
+                    {
+                        text: 'Okay',
+                        style: 'cancel',
+                    },
+                ],
+            );
+        });
+
+        it('shows alert for non-cloud starter', () => {
+            const mockAlert = jest.spyOn(Alert, 'alert');
+            showLimitRestrictedAlert({
+                limitRestricted: true,
+                maxParticipants: 8,
+                isCloudStarter: false,
+            }, intl as any);
+
+            expect(mockAlert).toHaveBeenCalledWith(
+                'This call is at capacity',
+                'The maximum number of participants per call is 8. Contact your System Admin to increase the limit.',
+                [
+                    {
+                        text: 'Okay',
+                        style: 'cancel',
+                    },
+                ],
+            );
+        });
+    });
+
+    describe('recordingAlert', () => {
+        beforeEach(() => {
+            needsRecordingAlert();
```

Comment:
For this file, iirc, we don't show the alerts again after showing once. Can we test for that as well? (unless we rare and I missed it)

Reply:
Let me check

Reply:
Indeed, we didn't cover that case. Added it, thanks!